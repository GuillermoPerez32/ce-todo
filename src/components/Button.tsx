import clsx from "clsx";

type Props = {
  variant: "outlined" | "contained";
  color: "primary" | "secondary";
  label: string;
  onClick?: () => void;
  icon?: string;
  disabled?: boolean;
  classname?: string;
};

export const Button = ({
  variant,
  color,
  label,
  onClick,
  icon,
  disabled,
  classname,
}: Props) => {
  return (
    <button
      className={clsx(
        "flex items-center justify-center rounded-md focus:outline-none py-1 px-3 gap-2 cursor-pointer transition-all " +
          classname,
        {
          "bg-blue-500 text-white":
            color === "primary" && variant === "contained",
          "bg-gray-200":
            (color === "secondary" && variant === "contained") ||
            (disabled && variant === "contained"),
          "bg-transparent border-2 border-gray-200": variant === "outlined",
          "text-gray-500": disabled,
        }
      )}
      onClick={onClick}
      disabled={disabled}
    >
      {icon && (
        <div
          className={clsx("transition-opacity", {
            "opacity-35": disabled,
          })}
        >
          <img src={icon} alt="icon" />
        </div>
      )}
      {label}
    </button>
  );
};
