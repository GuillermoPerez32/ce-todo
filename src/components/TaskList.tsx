import clsx from "clsx";
import { useTaskStore } from "../services/task";
import { renderFormattedText } from "../utils/renderFormattedText";

export const TaskList = () => {
  const tasks = useTaskStore((state) => state.tasks);
  const setTaskDone = useTaskStore((state) => state.setDone);
  const removeTask = useTaskStore((state) => state.remove);

  return (
    <div className="mt-6">
      {tasks.map((task) => (
        <div key={task.id} className="p-2 mt-1 flex items-center">
          <input
            type="checkbox"
            className="focus:ring-blue-500 h-4 w-4"
            checked={task.done}
            onChange={(e) => {
              setTaskDone(task.id, e.target.checked);
              setTimeout(() => {
                removeTask(task.id);
              }, 500);
            }}
          />
          <p
            className={clsx("ml-2 transition-opacity duration-500", {
              "line-through opacity-0": task.done,
            })}
          >
            {renderFormattedText(task.content)}
          </p>
        </div>
      ))}
    </div>
  );
};
