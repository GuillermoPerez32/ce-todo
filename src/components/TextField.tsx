import { useRef, useState } from "react";
import AddSvg from "../assets/plus-square.svg";
import OpenSvg from "../assets/maximize-2.svg";
import CalendarSvg from "../assets/calendar.svg";
import UnlockSvg from "../assets/unlock.svg";
import SunSvg from "../assets/sun.svg";
import CircleSvg from "../assets/circle.svg";
import { Button } from "./Button";
import clsx from "clsx";

export const TextField = () => {
  const [text, setText] = useState("");
  const [focused, setFocused] = useState<boolean>(false);
  const inputRef = useRef<HTMLInputElement>(null);

  const isDisabled = text === "";

  return (
    <div className="pt-14 px-14">
      <div
        className={`w-full flex items-center rounded-md border px-4 pb-8 pt-2 ${
          focused ? "border-gray-200 shadow" : "border-transparent"
        }`}
      >
        <img src={AddSvg} alt="add task" />
        <input
          ref={inputRef}
          value={text}
          onChange={(e) => setText(e.target.value)}
          type="text"
          className="focus:outline-none px-2 w-full h-full"
          placeholder="Type to add a new task"
          onFocus={() => {
            setFocused(true);
          }}
        />
      </div>
      <div
        className={clsx("p-2 shadow mt-1 flex", {
          hidden: !focused,
        })}
      >
        <Button
          classname="ml-4"
          variant="contained"
          color="secondary"
          label="Open"
          icon={OpenSvg}
          disabled={isDisabled}
        />
        <Button
          classname="ml-4"
          variant="outlined"
          color="secondary"
          label="Today"
          icon={CalendarSvg}
          disabled={isDisabled}
        />
        <Button
          classname="ml-1"
          variant="outlined"
          color="secondary"
          label="Public"
          icon={UnlockSvg}
          disabled={isDisabled}
        />
        <Button
          classname="ml-1"
          variant="outlined"
          color="secondary"
          label="Highlight"
          icon={SunSvg}
          disabled={isDisabled}
        />
        <Button
          classname="ml-1"
          variant="outlined"
          color="secondary"
          label="Estimation"
          icon={CircleSvg}
          disabled={isDisabled}
        />
        <Button
          classname="ml-auto"
          variant="contained"
          color="secondary"
          label="Cancel"
          onClick={() => {
            setFocused(false);
            setText("");
          }}
        />
        <Button
          classname="ml-1 mr-4"
          variant="contained"
          color="primary"
          label="Ok"
          disabled={isDisabled}
        />
      </div>
    </div>
  );
};
