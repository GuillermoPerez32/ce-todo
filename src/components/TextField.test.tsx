import { test, describe, expect } from "vitest";
import { TextField } from "../components/TextField";
import { render } from "@testing-library/react";

describe("<TextField />", () => {
  test("renders TextField with label", () => {
    const { container } = render(<TextField />);
    expect(container).toMatchSnapshot();
  });
});
