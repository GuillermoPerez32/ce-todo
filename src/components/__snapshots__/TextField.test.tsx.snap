// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TextField /> > renders TextField with label 1`] = `
<div>
  <div
    class="pt-14"
  >
    <div
      class="w-full flex items-center rounded-md border px-4 pb-8 pt-2 border-transparent"
    >
      <img
        alt="add task"
        src="/src/assets/plus-square.svg"
      />
      <input
        class="focus:outline-none px-2 w-full h-full"
        placeholder="Type to add a new task"
        type="text"
        value=""
      />
    </div>
    <div
      class="p-2 shadow mt-1 flex hidden"
    >
      <button
        class="flex items-center justify-center rounded-md focus:outline-none py-1 px-3 gap-2 cursor-pointer transition-all ml-4 bg-gray-200 text-gray-500"
        disabled=""
      >
        <div
          class="transition-opacity opacity-35"
        >
          <img
            alt="icon"
            src="/src/assets/maximize-2.svg"
          />
        </div>
        Open
      </button>
      <button
        class="flex items-center justify-center rounded-md focus:outline-none py-1 px-3 gap-2 cursor-pointer transition-all ml-4 bg-transparent border-2 border-gray-200 text-gray-500"
        disabled=""
      >
        <div
          class="transition-opacity opacity-35"
        >
          <img
            alt="icon"
            src="/src/assets/calendar.svg"
          />
        </div>
        Today
      </button>
      <button
        class="flex items-center justify-center rounded-md focus:outline-none py-1 px-3 gap-2 cursor-pointer transition-all ml-1 bg-transparent border-2 border-gray-200 text-gray-500"
        disabled=""
      >
        <div
          class="transition-opacity opacity-35"
        >
          <img
            alt="icon"
            src="/src/assets/unlock.svg"
          />
        </div>
        Public
      </button>
      <button
        class="flex items-center justify-center rounded-md focus:outline-none py-1 px-3 gap-2 cursor-pointer transition-all ml-1 bg-transparent border-2 border-gray-200 text-gray-500"
        disabled=""
      >
        <div
          class="transition-opacity opacity-35"
        >
          <img
            alt="icon"
            src="/src/assets/sun.svg"
          />
        </div>
        Highlight
      </button>
      <button
        class="flex items-center justify-center rounded-md focus:outline-none py-1 px-3 gap-2 cursor-pointer transition-all ml-1 bg-transparent border-2 border-gray-200 text-gray-500"
        disabled=""
      >
        <div
          class="transition-opacity opacity-35"
        >
          <img
            alt="icon"
            src="/src/assets/circle.svg"
          />
        </div>
        Estimation
      </button>
      <button
        class="flex items-center justify-center rounded-md focus:outline-none py-1 px-3 gap-2 cursor-pointer transition-all ml-auto bg-gray-200"
      >
        Cancel
      </button>
      <button
        class="flex items-center justify-center rounded-md focus:outline-none py-1 px-3 gap-2 cursor-pointer transition-all ml-1 mr-4 bg-blue-500 text-white bg-gray-200 text-gray-500"
        disabled=""
      >
        Ok
      </button>
    </div>
  </div>
</div>
`;
